pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -116,19 +127,47 @@ func (h *appHandler) apply(ctx context.Context, appRev *v1beta1.ApplicationRevis
 
 	for _, comp := range comps {
 		comp.SetOwnerReferences(owners)
+		needTracker, err := h.checkResourceNeedResourceTracker(ctx, &comp.Spec.Workload)
+		if err != nil {
+			return err
+		}
 		newComp := comp.DeepCopy()
 		// newComp will be updated and return the revision name instead of the component name
 		revisionName, err := h.createOrUpdateComponent(ctx, newComp)
 		if err != nil {
 			return err
 		}
+		if needTracker {
+			workloadName, err := h.getWorkloadName(comp.Spec.Workload, comp.Name, revisionName)
+			if err != nil {
+				return err
+			}
+			if err = h.recodeTrackedResource(workloadName, newComp.Spec.Workload); err != nil {
+				return err
+			}
+		}
```

Comment:
why not merge the logic into one function called `handleResourceTracker`?

```
comp.SetOwnerReferences(owners)
newComp := comp.DeepCopy()
// newComp will be updated and return the revision name instead of the component name
revisionName, err := h.createOrUpdateComponent(ctx, newComp)
if err != nil {
	return err
}
h.handleResourceTracker(...)
```